src/main/groovy/org/mockito/release/internal/gradle/GitStatusPlugin.java

```diff
@@ -0,0 +1,53 @@
+package org.mockito.release.internal.gradle;
+
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.mockito.release.exec.Exec;
+import org.mockito.release.exec.ProcessRunner;
+
+public class GitStatusPlugin implements Plugin<Project> {
+
+    private GitStatus gitStatus;
+
+    @Override
+    public void apply(Project project) {
+        if (project.getParent() == null) {
+            gitStatus = project.getExtensions().create(GitPlugin.class.getName(), GitStatus.class, project);
+        } else {
+            gitStatus = project.getRootProject().getExtensions().create(GitPlugin.class.getName(), GitStatus.class, project);
+        }
+    }
+
+    public GitStatus getGitStatus() {
+        return gitStatus;
+    }
+
+    public static class GitStatus {
+
+        private final ProcessRunner processRunner;
+        private volatile String branchName;
+        private static final Object SYNC = new Object();
+
+        public GitStatus(Project project) {
+             processRunner = Exec.getProcessRunner(project.getRootDir());
+        }
+
+        public void setBranchName(String branchName) {
```

Comment:
Let's make it even simpler :)

1. I suggest we get rid of the 'setBranchName' method from here
2. We can create a new task type 'GitCheckOutTask' task that has 'setRev' method which is the target revision we want to check out.
3. New task type 'GitCheckOutTask' will be used for creating 'checkOutBranch' task  in GitPlugin
4. TravisPlugin already applies GitPlugin so travis plugin can configure checkOutBranch.setRev with the value acquired from env variable

This way, we don't offer way to set branch and things are simpler.